server/src-lib/Hasura/RQL/DDL/RemoteSchema.hs

```diff
@@ -123,12 +169,26 @@ buildGCtxMap = do
   GS.buildGCtxMapPG
   sc <- askSchemaCache
   let gCtxMap = scGCtxMap sc
-  -- Stitch remote schemas
-  (mergedGCtxMap, defGCtx) <- mergeSchemas (scRemoteSchemas sc) gCtxMap
+  -- add remote schemas
+  mergedGCtxMap <- mergeRemoteSchemas (scRemoteSchemas sc) gCtxMap
+  allRemotesGCtx <- mkAllRemoteGCtx (scRemoteSchemas sc)
   writeSchemaCache sc { scGCtxMap = mergedGCtxMap
-                      , scDefaultRemoteGCtx = defGCtx
+                      , scAllRemoteGCtx = allRemotesGCtx
                       }
 
+mkCacheForRemoteSchema
+  :: (QErrM m) => SchemaCache -> RemoteSchemaName -> m SchemaCache
+mkCacheForRemoteSchema sc rsName = do
```

Comment:
Can we have this as `RemoteSchemaMap -> RemoteSchemaName -> m SchemaCache`, which has less confusion and make sense for the function name it has.